core/consensus/aura/slots/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+extern crate parity_codec as codec;
+
+mod slots;
+
+pub use slots::{Slots, SlotInfo};
+
+use std::sync::{mpsc, Arc};
+use std::thread;
+use futures::prelude::*;
+use futures::{Future, IntoFuture, future::{self, Either}};
+use log::{warn, debug, info};
+use runtime_primitives::generic::BlockId;
+use runtime_primitives::traits::{ProvideRuntimeApi, Block};
+use consensus_common::SyncOracle;
+use inherents::{InherentData, InherentDataProviders};
+use aura_primitives::AuraApi;
+use client::ChainHead;
+use codec::Encode;
+
+/// A worker that should be invoked at every new slot.
+pub trait SlotWorker<B: Block> {
+	/// Called when the proposer starts.
+	fn on_start(
+		&self,
+		inherent_data_providers: &InherentDataProviders,
+		slot_duration: u64
+	) -> Result<(), consensus_common::Error>;
+
+	/// Called when a new slot is triggered.
+	fn on_slot(
+		&self,
+		chain_head: B::Header,
+		slot_info: SlotInfo,
+	) -> Box<Future<Item=(), Error=consensus_common::Error>>;
+}
+
+/// Slot compatible inherent data.
+pub trait SlotCompatible {
```

Comment:
I don't think that we require this. The `InherentData` type is always the same for the complete core.

Reply:
But we can't be sure that they'll use the same inherent data type. Our current timestamp and slot are all specific to srml.

Reply:
But if you want to use `Aura` you probably need to use the srml module anyway?